examples/moonolith/ex1p.cpp

```diff
@@ -0,0 +1,245 @@
+                  "Number of src refinements");
+   args.AddOption(&dest_n_refinements, "-dr", "--dest_refinements",
+                  "Number of dest refinements");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&use_vector_fe, "-vfe", "--use_vector_fe", "-no-vfe",
+                  "--no-vector_fe", "Use vector finite elements");
+
+   args.Parse();
+   check_options(args);
+
+   ///////////////////////////////////////////////////
+   shared_ptr<Mesh> src_mesh, dest_mesh;
+
+   ifstream imesh;
+
+   imesh.open(destination_mesh_file);
+   if (imesh)
+   {
+      dest_mesh = make_shared<Mesh>(imesh, 1, 1);
+      imesh.close();
+   }
+   else
+   {
+      if (rank == 0)
+         mfem::err << "WARNING: Destination mesh file not found: "
+                   << destination_mesh_file << "\n"
+                   << "Using default 2D quad mesh.";
+
+      dest_mesh = make_shared<Mesh>(4, 4, Element::QUADRILATERAL);
+   }
+
+   const int dim = dest_mesh->Dimension();
+
+   dest_mesh->Transform(&destination_transform);
+
+   Vector box_min(dim), box_max(dim), range(dim);
+   dest_mesh->GetBoundingBox(box_min, box_max);
+   range = box_max;
+   range -= box_min;
+
+   imesh.open(source_mesh_file);
+
+   if (imesh)
+   {
+      src_mesh = make_shared<Mesh>(imesh, 1, 1);
+      imesh.close();
+   }
+   else
+   {
+      if (rank == 0)
+         mfem::err << "WARNING: Source mesh file not found: " << source_mesh_file
+                   << "\n"
+                   << "Using default box mesh.\n";
+
+      if (dim == 2)
+      {
+         src_mesh =
+            make_shared<Mesh>(4, 4, Element::TRIANGLE, 1, range[0], range[1]);
+      }
+      else if (dim == 3)
+      {
+         src_mesh = make_shared<Mesh>(4, 4, 4, Element::TETRAHEDRON, 1, range[0],
+                                      range[1], range[2]);
+      }
+
+      for (int i = 0; i < src_mesh->GetNV(); ++i)
+      {
+         double *v = src_mesh->GetVertex(i);
+
+         for (int d = 0; d < dim; ++d)
+         {
+            v[d] += box_min[d];
+         }
+      }
+   }
+
+   for (int i = 0; i < src_n_refinements; ++i)
+   {
+      src_mesh->UniformRefinement();
+   }
+
+   for (int i = 0; i < dest_n_refinements; ++i)
+   {
+      dest_mesh->UniformRefinement();
+   }
+
+   auto p_src_mesh = make_shared<ParMesh>(MPI_COMM_WORLD, *src_mesh);
+   auto p_dest_mesh = make_shared<ParMesh>(MPI_COMM_WORLD, *dest_mesh);
+
+   ///////////////////////////////////////////////////
+
+   shared_ptr<FiniteElementCollection> src_fe_coll, dest_fe_coll;
+
+   if (use_vector_fe)
+   {
+      src_fe_coll = make_shared<RT_FECollection>(1, src_mesh->Dimension());
+      dest_fe_coll = make_shared<RT_FECollection>(1, dest_mesh->Dimension());
+   }
+   else
+   {
+      src_fe_coll = make_shared<L2_FECollection>(1, src_mesh->Dimension());
```

Comment:
The two FE orders should be a command line option.

Reply:
Done.